src/charts/tau.gpl.js

```diff
@@ -0,0 +1,241 @@
+            left     : 0,
+            top      : 0,
+            width    : size.width,
+            height   : size.height
+        };
+
+        this.drawUnitsStructure(this.root);
+    }
+
+    expandUnitsStructure(rootUnit) {
+
+        var buildRecursively = (root, parentPipe) => {
+
+            if (root.expression.operator !== false) {
+
+                var expr = this.parseExpression(root.expression, parentPipe);
+
+                root.transformation = root.transformation || [];
+
+                root.frames = expr.exec().map((tuple) => {
+
+                    var pipe = parentPipe
+                        .concat([
+                            {
+                                type: 'where',
+                                args: tuple
+                            }
+                        ])
+                        .concat(root.transformation);
+
+                    var item = {
+                        source: expr.source,
+                        pipe: pipe
+                    };
+
+                    if (tuple) {
+                        item.key = tuple;
+                    }
+
+                    item.unit = (root.unit) ? root.unit.map((unit) => utils.clone(unit)) : [];
+
+                    return item;
+                });
+            }
+
+            root.frames.map((item) => {
+                // key: tuple,
+                // source: expr.source,
+                // pipe: pipe
+
+                item.unit.map((unit) => buildRecursively(unit, item.pipe));
+
+                return item;
+            });
+
+            return root;
+        };
+
+        return buildRecursively(rootUnit, []);
+    }
+
+    drawUnitsStructure() {
+
+        var continueDrawRecursively = (rootConf, rootFrame) => {
+
+            var dataFrame = this.datify(calcBaseFrame(rootConf.expression, rootFrame));
+
+            var UnitClass = this.unitSet.get(rootConf.type);
+
+            var unitNode = new UnitClass(rootConf);
+
+            unitNode
+                .drawLayout((type, alias, settings) => {
+
+                    // type is one of:
+                    // - pos
+                    // - size
+                    // - color
+                    // - shape
+                    var name = alias ? alias : `${type}:default`;
+
+                    return this.scalesCreator.create(this.scales[name], dataFrame, settings);
+                })
+                .drawFrames(rootConf.frames.map(this.datify.bind(this)), continueDrawRecursively);
+
+            return rootConf;
+        };
+
+        return continueDrawRecursively(this.root);
+    }
+
+    datify(frame) {
+        var data = this.sources[frame.source].data;
+        var trans = this.trans;
+        frame.take = () => frame.pipe.reduce((data, pipeCfg) => trans[pipeCfg.type](data, pipeCfg.args), data);
+        frame.data = frame.take();
+        return frame;
+    }
+
+    parseExpression(expr, parentPipe) {
```

Comment:
You can use Es6 parseExpression({operator='none', source,inherit,params}, parentPipe)